### DISCUSSION TITLE
Can you force bun tests to run sequentially? (integration tests)

### DISCUSSION THREAD

**Discussion by mbbirkel** [oven-sh/bun#4831] [Q&A]
Is it possible to force bun to run tests sequentially?

My tests are dependent on database state, and some fail when running `bun test`. I suspect this is caused by race conditions from running tests concurrently. When i run the failing tests in isolation, however, they pass.

Running the tests with Jest works fine, although execution is a lot slower.

**Comment by dweldon** [comment]
As of this writing, I'm using bun `1.1.34`, and it seems like the tests all run sequentially by default. I have a suite of integration tests in several files and they don't appear to be causing a conflict. Note that I'm resetting the state of the DB after each tests completes so there are no inter-dependencies.

If this assumption about sequential execution is incorrect, someone please let me know.

**Comment by krisbitney** [comment]
While the docs do say that tests should run sequentially, they are not running sequentially. I am observing race conditions due to my tests being dependent on database state as well.

**Reply by francescov1** [reply]
@krisbitney we've been finding the same, especially notice it in more resource constrained environments
